ToolBox/GDAXDownloader/GDAXDownloader.cs

```diff
@@ -0,0 +1,114 @@
+using QuantConnect.Data.Market;
+using Newtonsoft.Json;
+using System.Linq;
+using System.Text;
+
+namespace QuantConnect.ToolBox.GDAXDownloader
+{
+    /// <summary>
+    /// Cryptoiq Data Downloader class 
+    /// </summary>
+    public class GDAXDownloader : IDataDownloader
+    {
+        private readonly string _exchange;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CryptoiqDownloader"/> class
+        /// </summary>
+        /// <param name="exchange">The bitcoin exchange</param>
+        /// <param name="scaleFactor">Scale factor used to scale the data, useful for changing the BTC units</param>
+        public GDAXDownloader(string exchange = Market.GDAX)
+        {
+            _exchange = exchange;
+        }
+
+        /// <summary>
+        /// Get historical data enumerable for a single symbol, type and resolution given this start and end time (in UTC).
+        /// </summary>
+        /// <param name="symbol">Symbol for the data we're looking for.</param>
+        /// <param name="resolution">Only Tick is currently supported</param>
+        /// <param name="startUtc">Start time of the data in UTC</param>
+        /// <param name="endUtc">End time of the data in UTC</param>
+        /// <returns>Enumerable of base data for this symbol</returns>
+        public IEnumerable<BaseData> Get(Symbol symbol, Resolution resolution, DateTime startUtc, DateTime endUtc)
+        {
+            List<BaseData> returnData = new List<BaseData>();
+
+            if (resolution != Resolution.Hour)
+            {
+                throw new ArgumentException("Only hourly data is currently supported.");
+            }
+
+            var counter = startUtc;
+            const string url = "http://api.gdax.com/products/{0}/candles?start={1}&end={2}&granularity={3}";
+
+            while (counter <= endUtc)
+            {
+                System.Threading.Thread.Sleep(1000);
+                Console.WriteLine("Getting " + counter.ToShortDateString() + " data..");
+                DateTime endDate = counter.AddDays(1);
+                int granularity = 3600;
+
+                var requestURL = string.Format(url, symbol.Value, counter.ToShortDateString(), endDate.ToShortDateString(), granularity);
+
+                HttpWebRequest request = (HttpWebRequest)WebRequest.Create(requestURL);
+                request.UserAgent = ".NET Framework Test Client";
+
+                HttpWebResponse response = (HttpWebResponse)request.GetResponse();
+                var encoding = ASCIIEncoding.ASCII;
+
+                string data;
+                using (var reader = new System.IO.StreamReader(response.GetResponseStream(), encoding))
+                {
+                    data = reader.ReadToEnd();
+                }
+
+                data = data.Remove(0, 2);
+                data = data.Substring(0,data.Length - 2);
+
+                string[] bars = data.Split(new string[] { "],[" },StringSplitOptions.RemoveEmptyEntries);
```

Comment:
Can you use JSON.NET? Parsing the JSON manually seems too fragile.

Reply:
Thanks. have to check how to use JSON.NET to parse this kind of return data:

  | [[1498953600,222.72,242.13,230,235,1512.2072792299994],[1498867200,220.21,249,243.86,230.07,4865.78202293998],[1498780800,238.09,272,258.31,243.86,6538.636067629987]]]
-- | --